src/components/Survey/Survey.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { Redirect } from 'react-router-dom';
+
+import { Form, Box, Heading } from 'grommet';
+
+import { saveSurvey } from '../../services/api';
+import SurveyField from '../SurveyField/SurveyField';
+import SurveyButton from '../Button';
+
+const messages = {
+  required: 'This field is required'
+};
+
+class Survey extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: props.data,
+      submitted: false,
+      success: false,
+      error: null
+    };
+  }
+
+  renderSurvey() {
+    const { submitted, success, error, data } = this.state;
+    if (submitted) {
+      return (
+        <Redirect
+          to={{
+            pathname: '/final',
+            state: { success, error }
+          }}
+        />
+      );
+    }
+    if (!data) {
```

Comment:
What is the point of returning null here? Handle that with below Error message. Also it is good to perform such error check at the beginning of method, and then proceed with business logic.